pkg/gce-pd-csi-driver/controller.go

```diff
@@ -283,7 +283,6 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 
 	// Validate if disk already exists
 	existingDisk, err := gceCS.CloudProvider.GetDisk(ctx, gceCS.CloudProvider.GetDefaultProject(), volKey, gceAPIVersion)
-	diskTypeForMetric = metrics.GetDiskType(existingDisk)
```

Comment:
What happens if the disk type of the existing disk doesn't match the disk type being requested in the storageclass?

Reply:
The existing disk is being validated on 293 below 
```
err = gceCS.CloudProvider.ValidateExistingDisk
```
The function `ValidateExistingDisk` checks for disk type, disk size, location and other params.